src/shogun/kernel/LogKernel.cpp

```diff
@@ -11,46 +11,32 @@
 
 using namespace shogun;
 
-LogKernel::LogKernel(): Kernel(0), m_distance(NULL), m_degree(1.8)
+LogKernel::LogKernel(): Kernel(0)
 {
-	init();
+	SG_ADD(&m_degree, "degree", "Degree kernel parameter.", ParameterProperties::HYPER);
+	SG_ADD(&m_distance, "distance", "Distance to be used.", ParameterProperties::HYPER);
 }
 
 LogKernel::LogKernel(int32_t cache, float64_t degree, std::shared_ptr<Distance> dist)
-: Kernel(cache), m_distance(std::move(dist)), m_degree(degree)
+: LogKernel()
 {
-	init();
-	ASSERT(m_distance)
-	
+	set_cache_size(cache);
+	m_degree = degree;
+	m_distance = std::move(dist);
+	ASSERT(m_distance);	
 }
 
 LogKernel::LogKernel(std::shared_ptr<Features >l, std::shared_ptr<Features >r, float64_t degree, std::shared_ptr<Distance> dist)
-: Kernel(10), m_distance(std::move(dist)), m_degree(degree)
-{
-	init();
-	ASSERT(m_distance)
-	
-	init(std::move(l), std::move(r));
+: LogKernel(10, degree, std::move(dist))
+{	
+	Kernel::init(l,r);
+	ASSERT(m_distance->init(l,r));
+	init_normalizer();
```

Comment:
Where did this come from?

Reply:
removing its init(l,r) method (used only inside LogKernel constructor and pasted the code here.

Reply:
Ok, but you need to keep the init(l,r) method so I guess just leave it there

Reply:
Do we absolutely need to? I removed init(l,r) from other classes as well.

Reply:
Yes
https://github.com/shogun-toolbox/shogun/blob/7014676a95ec27b51473f037c4c137083075d61e/src/shogun/kernel/Kernel.h#L455

Reply:
Actually it isn’t pure virtual so I guess you don’t need to

Reply:
Very well, so I'll restore the init(l,r) methods. Maybe we can do something about them in future

Reply:
> Actually it isn’t pure virtual so I guess you don’t need to

Yes but there are some methods (other than constructors) which are calling init(l,r).

Reply:
Ah ok, then yes keep it